corehq/motech/repeaters/utils.py

```diff
@@ -3,57 +3,12 @@
 from corehq.apps.cleanup.management.commands.populate_sql_model_from_couch_model import (
     PopulateSQLCommand,
 )
-from corehq.motech.dhis2.repeaters import (
-    SQLDhis2EntityRepeater,
-    SQLDhis2Repeater,
-)
-from corehq.motech.fhir.repeaters import SQLFHIRRepeater
 from corehq.motech.models import ConnectionSettings
-from corehq.motech.openmrs.repeaters import SQLOpenmrsRepeater
 from corehq.motech.repeaters.dbaccessors import (
     get_all_repeater_docs,
     get_repeater_count_for_domains,
     get_repeaters_by_domain,
 )
-from corehq.motech.repeaters.expression.repeaters import (
-    SQLCaseExpressionRepeater,
-)
-from corehq.motech.repeaters.models import (
-    SQLAppStructureRepeater,
-    SQLCaseRepeater,
-    SQLCreateCaseRepeater,
-    SQLDataRegistryCaseUpdateRepeater,
-    SQLFormRepeater,
-    SQLLocationRepeater,
-    SQLReferCaseRepeater,
-    SQLShortFormRepeater,
-    SQLUpdateCaseRepeater,
-    SQLUserRepeater,
-)
-from custom.cowin.repeaters import (
-    SQLBeneficiaryRegistrationRepeater,
-    SQLBeneficiaryVaccinationRepeater,
```

Comment:
I have two thoughts on that, 
1. Since `RepeaterMeta` would be populating the `REPEATER_CLASS_MAP` automatically as soon as it encounters a new class. So is it required to have it populated on the startup itself, if not then do we need them imported?
2. We already use `REPEATER_CLASSES` in [settings.py ](https://github.com/dimagi/commcare-hq/blob/master/settings.py#L813-L831). Can we rely on it to populate the map?

Reply:
> 1. Since `RepeaterMeta` would be populating the `REPEATER_CLASS_MAP` automatically as soon as it encounters a new class. So is it required to have it populated on the startup itself, if not then do we need them imported?

I think we want to have `REPEATER_CLASS_MAP` fully populated after `RepeaterAppConfig.ready()` is called

https://github.com/dimagi/commcare-hq/blob/b24dfe162ee49992c0c345517e5bb7bde7ea1698/corehq/motech/repeaters/apps.py#L7

That way any code that needs to lookup a class in that mapping will be able to do it safely and deterministically (not based on whether some module has been imported or not yet). As I'm thinking about this more, rather than having these imports be done in this file (corehq/motech/repeaters/utils.py) they should be done in `RepeaterAppConfig.ready()` if possible.

> 2. We already use `REPEATER_CLASSES` in [settings.py ](https://github.com/dimagi/commcare-hq/blob/master/settings.py#L813-L831). Can we rely on it to populate the map?

Maybe, although we still need some a test to verify that the `REPEATER_CLASS_MAP` actually contains all of the repeater classes that it should after `RepeaterAppConfig.ready()`.

Reply:
Understood. Thanks for explaining. I think I have some idea about making it work. I will make the changes.